added hep17 folder, 2d eff plots for egammaHLT DQM and Validation

Monophoton opened the pull request:
- Added HEP17 folder with plots in /HLT/EgOffline/ folder similar to HLT/EgOffline/Source_Histos/ but for HEP17, HEM17 region and ratio plots.
- Also added 2D efficiency plot (EtaVsPhi) per filter efficiency in HLT/HLTEgammaValidation/ directories.
- a minor bug fix (not used in code)

sushilchauhan committed:
added hep17 folder, 2d eff plots for egammaHLT DQM and Validation

cmsbuild commented:
A new Pull Request was created by  @Monophoton  for master.

It involves the following packages:

DQMOffline/Trigger
HLTriggerOffline/Egamma


@vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@battibass, @mtosi, @jhgoh, @calderona, @HuguesBrun, @trocino, @rociovilar this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

mtosi commented on DQMOffline/Trigger/interface/EgHLTEgCutCodes.h:
```diff
@@ -46,11 +46,11 @@ namespace egHLT {
+      HLTDETAIN     =0x02000000,
+      HLTDPHIIN     =0x04000000,
+      HLTINVEINVP   =0x08000000,
       //flag that if its set, shows the code is invalid
```
are you sure ?

Monophoton replied:
otherwise they were similar to:
HLTISOLTRKSPHO=0x00020000,
HLTISOLHAD =0x00040000,
HLTISOLEM =0x00080000
I ask Gabi and she also thinks its a bug.

mtosi commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

mtosi commented:
i simply saw a difference wrt previous version of the code
and i would make sure that such update is intended 

thanks

> On 12 Jul 2017, at 13:28, Monophoton <notifications@github.com> wrote:
> 
> @Monophoton commented on this pull request.
> 
> In DQMOffline/Trigger/interface/EgHLTEgCutCodes.h:
> 
> >        //hlt quantities that are slightly different to reco
> -      HLTDETAIN     =0x00020000,
> -      HLTDPHIIN     =0x00040000,
> -      HLTINVEINVP   =0x00080000,
> +      HLTDETAIN     =0x02000000,
> +      HLTDPHIIN     =0x04000000,
> +      HLTINVEINVP   =0x08000000,
>        //flag that if its set, shows the code is invalid
> otherwise they were similar to:
> HLTISOLTRKSPHO=0x00020000,
> HLTISOLHAD =0x00040000,
> HLTISOLEM =0x00080000
> I ask Gabi and she also thinks its a bug.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

cmsbuild commented:
+1
Tested at: a3416213e57ca7cc9f1f49fdb2460b33768eb4fe
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/21395/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/de56eaa6a192fe2ad85b6d0f065f546ed7861186
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/21395/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/21395/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/21395/summary.html

Comparison Summary:
 - You potentially added 58 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 2015567
 - DQMHistoTests: Total failures: 15044
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2000357
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 93 log files, 14 edm output root files, 23 DQM output files

dmitrijus commented on HLTriggerOffline/Egamma/src/EmDQMPostProcessor.cc:
```diff
@@ -359,6 +368,40 @@ TProfile* EmDQMPostProcessor::dividehistos(DQMStore::IBooker & ibooker, DQMStore
+
+  MonitorElement* meOut = ibooker.book2D(outName,titel,num->GetXaxis()->GetNbins(),num->GetXaxis()->GetXmin(),num->GetXaxis()->GetXmax(),num->GetYaxis()->GetNbins(),num->GetYaxis()->GetXmin(),num->GetYaxis()->GetXmax());
+ TH2F* out= meOut->getTH2F();
+  std::cout<<num->GetXaxis()->GetNbins()<<" ,"<<num->GetXaxis()->GetXmin()<<" ,"<<num->GetXaxis()->GetXmax()<<" , "<<num->GetYaxis()->GetNbins()<<" , "<<num->GetYaxis()->GetXmin()<<" , "<<num->GetYaxis()->GetXmax()<<std::endl;
```
Using std::cout is forbidden! Use LogInfo/LogWarning!

sushilchauhan committed:
removed std:cout from EmDQMPostProcessor

cmsbuild commented:
Pull request #19699 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

Monophoton commented:
that std::cout was by mistake and left over from a test. Update the commit to remove it.

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

cmsbuild commented:
+1
Tested at: b7ea6795130176107f3ba824226bfa56c49bc59d
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/21474/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/2136e028f24d4e387afda113972d080c5789805d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/21474/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/21474/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/21474/summary.html

Comparison Summary:
 - You potentially added 50 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 2022761
 - DQMHistoTests: Total failures: 14931
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2007664
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 93 log files, 14 edm output root files, 23 DQM output files

Monophoton commented:
something more need to be done from my side?

davidlange6 commented:
hi @Monophoton - could you rebase this - thanks

Monophoton commented:
@davidlange6, I never did this, could you please give me the step to rebase it? Thanks!

davidlange6 commented:
http://cms-sw.github.io/tutorial-resolve-conflicts.html#backporting-a-pr

> On Jul 20, 2017, at 11:43 AM, Sushil S. Chauhan <notifications@github.com> wrote:
> 
> @davidlange6, I never did this, could you please give me the step to rebase it? Thanks!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

Monophoton commented:
@davidlange6 , thanks for the link. Just one minor thing to ask, what will be the new base here (old base is CMSSW_9_3_X_2017-07-11-2300). Thanks.

mtosi commented:
CMSSW_9_3_X should work, shouldn't it ?

do you have the PR w/ the backport for 92x ?
thanks !

Monophoton commented:
@davidlange6 Hi I run following steps. Just to avoid any mess please confirm this is okay.
cmsrel CMSSW_9_3_0_pre1 
cmsenv
git cms-checkout-topic -u sushilchauhan:egammaHLTOffDQM_HEP17_2DEffPlots

git rebase --onto CMSSW_9_3_0_pre1 CMSSW_9_3_X_2017-07-11-2300 egammaHLTOffDQM_HEP17_2DEffPlots

git diff (and then resolve confict in ExoticaMonitoring_Client_cff.py)
git add DQMOffline/Trigger/python/ExoticaMonitoring_Client_cff.py
git rebase --continue

I haven't push it to my-cmss branch yet but will do once confirmed this is okay.
Thanks

Monophoton commented:
Hi, If some experts could confirm the above rebasing steps are okay then I could push it to my branch to update this pull request.  Thanks a lot.

Monophoton commented:
I will appreciate a quick response to my query so that we could finalize this PR (it has been 2 weeks now since I submitted it). I could submit one with 92X today also. Thanks.

dmitrijus commented on DQMOffline/Trigger/python/DQMOffline_Trigger_cff.py:
```diff
@@ -81,11 +81,17 @@
     lumiMonitorHLTsequence *
     hcalMonitoringSequence *
-    egHLTOffDQMSource *
+    (egHLTOffDQMSource+egHLTOffDQMSource_HEP17) *
```
Why not just add a single line to the sequence?
```
egHLTOffDQMSource *
egHLTOffDQMSource_HEP17 * 
```

dmitrijus commented:
Hi, I don't know if those commands will help you.

My favourite way to rebase:
1. cmsrel/scram p the latest ib
2. git cherry-pick <commit id>
3. repeat step 2 for every commit you did, in order.
4. compile, test and force push to the **same** branch.

This PR should update as soon as you push. 

And instead of blinding using the commands, I would suggest you to familiarize yourself with git.
Typing "git rebase" or even "git basics" in your favourite search engine should give you enough material...

Also, you are touching the common Trigger files. I suggest you coordinate your activities with Andrea (@fwyzard).

Monophoton commented:
Hi @dmitrijus , I followed the example here (http://cms-sw.github.io/tutorial-resolve-conflicts.html#backporting-a-pr). 
My confusion was that the new base (CMSSW_9_3_0_pre1 ) i choose  is ok  or  not  as I don't find any IB of 93X when I do scram list CMSSW.

dmitrijus commented:
> My confusion was that the new base (CMSSW_9_3_0_pre1 )

CMSSW_9_3_0_pre1 is too old. Use IB!

>  i choose is ok or not as I don't find any IB of 93X when I do scram list CMSSW.

Make sure you have a proper SCRAM_ARCH set.

Monophoton commented:
Hi @dmitrijus the SCRAM_ARCH is set to slc6_amd64_gcc530 and I thought this is the right one(?) but I don't see any IB of 93X with this when I do scram list CMSSW.

smuzaffar commented:
@Monophoton , ```scram list CMSSW_9_3_X``` should list all IBs for available archs. production arch for 93X is slc6_amd64_gcc630

Monophoton commented:
okay. I retried but it gives error at this step: 
[lxplus084@src] git rebase --onto CMSSW_9_3_X_2017-07-28-2300 CMSSW_9_3_X_2017-07-11-2300 egammaHLTOffDQM_HEP17_2DEffPlots

fatal: Needed a single revision
invalid upstream CMSSW_9_3_X_2017-07-11-2300

looks like I am doing something wrong  here.

fwyzard commented:
I think you need to give a branch, not a tag, as the argument to rebase.
At least, this command works for me:
```
git rebase --onto CMSSW_9_3_X_2017-07-28-2300 official-cmssw/CMSSW_9_3_X egammaHLTOffDQM_HEP17_2DEffPlots
```
of course then you'll need to fix the conflicts in
       both modified:      DQMOffline/Trigger/interface/EgHLTPhoHLTFilterMon.h
       both modified:      DQMOffline/Trigger/python/DQMOffline_Trigger_cff.py
       both modified:      DQMOffline/Trigger/python/ExoticaMonitoring_Client_cff.py
       both modified:      DQMOffline/Trigger/src/EgHLTEleHLTFilterMon.cc
       both modified:      DQMOffline/Trigger/src/EgHLTOfflineSource.cc
       both modified:      DQMOffline/Trigger/src/EgHLTPhoHLTFilterMon.cc

Monophoton commented:
thanks Andrea, yes I am working to resolve those conflicts.

sushilchauhan committed:
minor fix

cmsbuild commented:
Pull request #19699 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

Monophoton commented:
just updated the branch rebasing to latest IB. Now there are no conflicts.
@dmitrijus Also updated the changes you suggested for DQMOffline_Trigger_cff.py  file.

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

sushilchauhan commented:
@mtosi  PR#19969 is also created for 92X.

cmsbuild commented:
+1
Tested at: 4f7ab9883902ff3365eb56129fc56c458f2e8b9b
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/21893/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/8e59646c5dc2756e96963b11dceef23e96c92c1b
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/21893/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/21893/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/21893/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 2446803
 - DQMHistoTests: Total failures: 56320
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2390317
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 93 log files, 14 edm output root files, 23 DQM output files

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

sushilchauhan commented:
I hope this PR is in queue for merge....please let me know if something more need to be done

davidlange6 commented on HLTriggerOffline/Egamma/src/EmDQMPostProcessor.cc:
```diff
@@ -370,6 +412,17 @@ EmDQMPostProcessor::getHistogram(DQMStore::IBooker & ibooker, DQMStore::IGetter
+EmDQMPostProcessor::get2DHistogram(DQMStore::IBooker & ibooker, DQMStore::IGetter & igetter, const std::string &histoPath)
+{
+  MonitorElement *monElement = igetter.get(histoPath);
+  if (monElement != NULL)
```
hi @Monophoton - sorry for the belated review

  use nullptr not NULL

sushilchauhan replied:
ok will do in my next update

davidlange6 commented on HLTriggerOffline/Egamma/src/EmDQMPostProcessor.cc:
```diff
@@ -359,6 +368,39 @@ TProfile* EmDQMPostProcessor::dividehistos(DQMStore::IBooker & ibooker, DQMStore
+
+  if(!num || !denom) return 0;
+
+  MonitorElement* meOut = ibooker.book2D(outName,titel,num->GetXaxis()->GetNbins(),num->GetXaxis()->GetXmin(),num->GetXaxis()->GetXmax(),num->GetYaxis()->GetNbins(),num->GetYaxis()->GetXmin(),num->GetYaxis()->GetXmax());
```
here and elsewhere there are indentation problems. Please check that code in loops is indented relative to the code outside the loop (and code in the same loop/function has the same indentation)

sushilchauhan replied:
ok. Will do it in next update.

davidlange6 commented on HLTriggerOffline/Egamma/src/EmDQMPostProcessor.cc:
```diff
@@ -359,6 +368,39 @@ TProfile* EmDQMPostProcessor::dividehistos(DQMStore::IBooker & ibooker, DQMStore
+
+  // Check if histograms actually exist
+
+  if(!num || !denom) return 0;
```
compare pointers to nullptr.

davidlange6 commented on HLTriggerOffline/Egamma/src/EmDQMPostProcessor.cc:
```diff
@@ -234,7 +235,7 @@ void EmDQMPostProcessor::dqmEndJob(DQMStore::IBooker & ibooker, DQMStore::IGette
       // Get the L1 over gen filter first
       filterName2= total->GetXaxis()->GetBinLabel(1);
-	
+      std::string tempvar;
```
I don't understand the need for this variable?

davidlange6 commented on HLTriggerOffline/Egamma/src/EmDQMPostProcessor.cc:
```diff
@@ -226,6 +226,7 @@ void EmDQMPostProcessor::dqmEndJob(DQMStore::IBooker & ibooker, DQMStore::IGette
       varNames.push_back("et");
       varNames.push_back("eta"); 
       if (!noPhiPlots) varNames.push_back("phi"); 
+         varNames.push_back("etaphi");       
```
is this intended to be inside the if?

sushilchauhan replied:
no it should be outside of the loop.

dmitrijus replied:
please fix this, it is really really misleading.

davidlange6 commented on DQMOffline/Trigger/src/EgHLTEleHLTFilterMon.cc:
```diff
@@ -76,21 +86,22 @@ EleHLTFilterMon::EleHLTFilterMon(MonElemFuncs& monElemFuncs,const std::string& f
   for(auto & eleMonElem : eleMonElems_) delete eleMonElem;
   for(auto & eleFailMonElem : eleFailMonElems_) delete eleFailMonElem;
   for(auto & eleEffHist : eleEffHists_) delete eleEffHist;
+  if(!doHEP_){ 
```
assuming the pointers are initialized (which they should be if not already), this if is not needed.

sushilchauhan replied:
may be I am over protective  here...

davidlange6 commented on DQMOffline/Trigger/interface/EgHLTMonElemFuncs.h:
```diff
@@ -84,6 +88,10 @@ namespace egHLT {
     }
 
+    template<class T,typename varType> void addStdHistHEP( std::vector<MonElemManagerBase<T>*>& histVec,const std::string& name,const std::string& title,                                                                                       
+                                                       const BinData::Data1D& binData,varType (T::*varFunc)()const){
```
the new template looks to be the exact same thing as the old one?

davidlange6 replied:
ah - its a MonELemMgrHEP..:( got it

davidlange6 commented on DQMOffline/Trigger/src/EgHLTMonElemFuncs.cc:
```diff
@@ -95,6 +96,99 @@ void MonElemFuncs::initStdPhoHists(std::vector<MonElemManagerBase<OffPho>*>& his
 }
+
+
+void MonElemFuncs::initStdEleHistsHEP(std::vector<MonElemManagerBase<OffEle>*>& histVec,const std::string& filterName,const std::string& baseName,const BinData& bins)
```
so for all the fancy templating of this code - the design falls apart here.  all the if (hep).. can go away, it would seem, if MonElemFuncs had a derived class for hep17. Was that considered? All the ifs make the details of the monitoring code completely un-understandable imho.

sushilchauhan replied:
honestly I didn't think about it ... but there are other histograms and efficiency plots which are not needed for new folder(e.g. I removed di object plots from MonElemFuncs). The efficiency plots are currenlty "off" but in future those could be turned "on". So I find it simpler(adding those if() ) to do this way.

sushilchauhan committed:
minor changes

cmsbuild commented:
Pull request #19699 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pr-code-checks/PR-19699/55

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/jenkins-artifacts/pr-code-checks/PR-19699/55/git-diff.patch

In future, you can run `scram build code-checks` to apply code checks

sushilchauhan commented:
How could I apply the above patch? Or I should do it by hand to update the pr. Thanks.

sushilchauhan committed:
applied patch to resolve  code style issue

cmsbuild commented:
Pull request #19699 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pr-code-checks/PR-19699/75

mtosi commented:
@cmsbuild, please test

> On 12 Aug 2017, at 08:46, cmsbuild <notifications@github.com> wrote:
> 
> +code-checks
> 
> Logs: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pr-code-checks/PR-19699/75
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 125b3278f725dc5654301b37b5a70bf5f7595d5f


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/56f2f723d8eee5e34bf7ce5ac2e6871fdfa7886d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22245/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22245/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22245/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
136.731 step1
<pre>DAS Error</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/56f2f723d8eee5e34bf7ce5ac2e6871fdfa7886d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22245/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22245/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

sushilchauhan commented:
@mtosi  I guess the above DAS Error is not related with the code

fwyzard commented:
It is not related, no.

Ant it looks like there is a non-transient DAS problem, so let's wait a couple of days to re-test this.

smuzaffar commented:
please test
DAS should be ok now

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 125b3278f725dc5654301b37b5a70bf5f7595d5f
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22251/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/56f2f723d8eee5e34bf7ce5ac2e6871fdfa7886d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22251/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22251/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22251/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 25
 - DQMHistoTests: Total histograms compared: 2600817
 - DQMHistoTests: Total failures: 252
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2600386
 - DQMHistoTests: Total skipped: 179
 - DQMHistoTests: Total Missing objects: 0
 - Checked 103 log files, 14 edm output root files, 25 DQM output files

sushilchauhan commented:
please let me know if this is ok then I could update the PR for 92X. Thanks

mtosi commented:
@dmitrijus @davidlange6 do you approve this code ?
can you please sign this PR ?
thanks !

sushilchauhan committed:
minor fix for if() condition in EmDQMPostProcessor.cc

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
Pull request #19699 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19699/267

mtosi commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 61a3af044adbee901331ef931967183fca843772


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/dbd8653f98656d364a0000b5d4134ebd48491af1
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22444/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22444/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22444/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
4.22 step1
<pre>DAS Error</pre>1003.0 step1
<pre>DAS Error</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/dbd8653f98656d364a0000b5d4134ebd48491af1
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22444/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22444/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

sushilchauhan commented:
@mtosi seems test need to be redon due to DAS error

mtosi commented:
@cmsbuild, please test

On Thu, Aug 24, 2017 at 8:42 AM, Sushil S. Chauhan <notifications@github.com
> wrote:

> @mtosi <https://github.com/mtosi> seems test need to be redon due to DAS
> error
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19699#issuecomment-324547913>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt581ddKeu9l0ah_6sZk6P1qQnuM5Wkks5sbRtSgaJpZM4OVSku>
> .
>

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 61a3af044adbee901331ef931967183fca843772
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22459/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/84cbb9ab89388b30ba01555e9fc85b9db56ceb63
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22459/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22459/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/22459/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2653934
 - DQMHistoTests: Total failures: 221
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2653524
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

sushilchauhan commented:
Hi @dmitrijus , does the PR looks ok now?

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

sushilchauhan commented:
Anything more need to be done from my side for this PR?

mtosi commented:
@dmitrijus @davidlange6 could we merge this PR ?

mtosi commented:
ah, wait, I've just realized (sorry)
@sushilchauhan there is a conflict to resolve
please, rebase

sushilchauhan commented:
@mia, do  I need to just update this branch rebasing it to 94X or I need to create a new  one for 94X. I assumed its former  right?

mtosi commented:
I think you just need to do rebase w.r.t. 94x

On Wed, Sep 20, 2017 at 11:40 AM, Sushil S. Chauhan <
notifications@github.com> wrote:

> @mia <https://github.com/mia>, do I need to just update this branch
> rebasing it to 94X or I need to create a new one for 94X. I assumed its
> former right?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19699#issuecomment-330800814>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt589W8WvisqeHF0lE2WeTjgBy1rbvWks5skN2qgaJpZM4OVSku>
> .
>

cmsbuild commented:
The code-checks are being triggered in jenkins.

sushilchauhan commented:
@mtosi  I just updated the branch by rebasing it to latest  94X IB.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19699/844

cmsbuild commented:
Pull request #19699 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

mtosi commented:
@cmsbuild, please test

On Wed, Sep 20, 2017 at 12:26 PM, cmsbuild <notifications@github.com> wrote:

> +code-checks
>
> Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19699/844
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19699#issuecomment-330811445>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt581x0Zh8HS31RzLlrTIZ6EzRUQ71dks5skOhRgaJpZM4OVSku>
> .
>

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 3dd553704e179e55bdbcee8c1648067765c4914f
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/23113/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5e53c2b9539aa99597276fc2af455b7acffca3cf
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/23113/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/23113/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/23113/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2649068
 - DQMHistoTests: Total failures: 232
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2648647
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

mtosi commented:
ciao
I'm a little bit confused
why was not this PR merged yet ?

thanks

mtosi commented:
@davidlange6 @dmitrijus @slava77

mtosi commented:
@sushilchauhan , please fix the conflict
by rebasing w.r.t. most recent 94x

cmsbuild commented:
The code-checks are being triggered in jenkins.

sushilchauhan commented:
@mtosi I just updated it again with rebasing to latest 94X.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19699/1067

cmsbuild commented:
Pull request #19699 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

cmsbuild commented:
+1
Tested at: c8fbfae956dfc44fb656fdf6c9c99b4b8875785a
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/23424/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/575a47c1dc033097d7d5fbda7e2fe14f71ee98b9
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/23424/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/23424/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/23424/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2761357
 - DQMHistoTests: Total failures: 267
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2760903
 - DQMHistoTests: Total skipped: 187
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 10 edm output root files, 26 DQM output files

sushilchauhan commented:
something more need to be done from my side?

davidlange6 commented:
please test
(I think we mainly need to confirm the old tests are stillok)

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: c8fbfae956dfc44fb656fdf6c9c99b4b8875785a
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/23786/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/8c0f84b67931b1b2929e779f34557ba0aae32c6f
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/23786/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/23786/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19699/23786/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2767139
 - DQMHistoTests: Total failures: 99
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2766869
 - DQMHistoTests: Total skipped: 171
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 10 edm output root files, 26 DQM output files

davidlange6 commented:
merge

mtosi commented:
@sushilchauhan, you need the backport in 92x, don't you ?

sushilchauhan commented:
@mtosi Yes (and I did it earlier), but i was wating to make sure there are no further udates. Right now I am travellling so will do it tomorrow asap.

mtosi commented:
thanks !
 mia

On Tue, Oct 17, 2017 at 10:48 AM, Sushil S. Chauhan <
notifications@github.com> wrote:

> @mtosi <https://github.com/mtosi> Yes (and I did it earlier), but i was
> wating to make sure there are no further udates. Right now I am travellling
> so will do it tomorrow asap.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19699#issuecomment-337162712>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt58zaV2mIp2jmdc6Sb9HYkESEj7oIgks5stGnogaJpZM4OVSku>
> .
>
